src/Microsoft.ML.Featurizers/Common.cs

```diff
@@ -0,0 +1,189 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Text;
+using Microsoft.Win32.SafeHandles;
+
+namespace Microsoft.ML.Featurizers
+{
+    #region Native Function Declarations
+
+    #endregion
+
+    internal enum FitResult : byte
+    {
+        Complete = 1, Continue, ResetAndContinue
+    }
+
+    // Not all these types are currently supported. This is so the ordering will allign with the native code.
+    internal enum TypeId : uint
+    {
+        String = 1, SByte, Short, Int, Long, Byte, UShort,
+        UInt, ULong, Float16, Float32, Double, Complex64,
+        Complex128, BFloat16, Bool, Timepoint, Duration,
+
+        LastStaticValue,
+        Tensor = 0x1001 | LastStaticValue + 1,
+        SparseTensor = 0x1001 | LastStaticValue + 2,
+        Tabular = 0x1001 | LastStaticValue + 3,
+        Nullable = 0x1001 | LastStaticValue + 4,
+        Vector = 0x1001 | LastStaticValue + 5,
+        MapId = 0x1002 | LastStaticValue + 6
+    };
+
+    [StructLayout(LayoutKind.Sequential, Pack = 1)]
+    internal unsafe struct NativeBinaryArchiveData
+    {
+        public byte* Data;
+        public IntPtr DataSize;
+    }
+
+    #region SafeHandles
+
+    internal class ErrorInfoSafeHandle : SafeHandleZeroOrMinusOneIsInvalid
+    {
+        [DllImport("Featurizers", EntryPoint = "DestroyErrorInfo", CallingConvention = CallingConvention.Cdecl), SuppressUnmanagedCodeSecurity]
```

Comment:
`Featurizers` seems like a pretty generic name for this native assembly. Is that the final name? Since it comes from the `Microsoft.MLFeaturizers` package, it may make sense to use `MLFeaturizers` as the native assembly name.

Reply:
That is the name as it currently stands. Closing as `ByDesign` for now but I will sync with the team and change this if the name gets updated.

---
In reply to: [340737413](https://github.com/dotnet/machinelearning/pull/4413#discussion_r340737413) [](ancestors = 340737413)